pkg/client/transport.go

```diff
@@ -68,6 +68,10 @@ func NewBearerAuthRoundTripper(bearer string, rt http.RoundTripper) http.RoundTr
 }
 
 func (rt *bearerAuthRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(req.Header.Get("Authorization")) != 0 {
```

Comment:
Not sure about this. Does basicAuthRoundTripper need it too? Maybe you should make a maybeAuth round tripper that can be inserted in the roundTripper stack?